base/loading.jl

```diff
@@ -1413,6 +1419,304 @@ function register_restored_modules(sv::SimpleVector, pkg::PkgId, path::String)
     return restored
 end
 
+# Restore backedges to external targets
```

Comment:
I think this moved code should go in a separate file from loading, since eventually most of it will become part of the Compiler

Reply:
Done.